### PR TITLE
fix(nuget): always coalesce sourceUrl and homepage if homepage available

### PR DESCRIPTION
**Pull request by NatMarchand**
## Changes

As suggested in #24843 when renovate is unable to fetch metadata from nuget, we don't want to early return in order to set sourceUrl with homepage if available.

## Context

When using github packages as nuget source, the call to fetch the nuspec always returns 404 as it is not implemented by github. Therefore, we're unable to get a value for sourceUrl and then cannot group our packages with `matchSourceUrls`

- Closes #24843 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by NatMarchand** [comment]
Please note that it changes current tests behavior, I can change the expected results accordingly.

File: lib/modules/datasource/nuget/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -1497,6 +1497,7 @@ exports[`modules/datasource/nuget/index getReleases processes real data (v3) nus
       "version": "3.12.0",
     },
   ],
+  "sourceUrl": "https://nunit.org/",
```
Comment (rarkins) [inline lib/modules/datasource/nuget/__snapshots__/index.spec.ts.snap:1500]: This is better suited for homepage  and not sourceUrl

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/datasource/nuget/__snapshots__/index.spec.ts.snap (hunk 1 above)
Comment (NatMarchand) [reply lib/modules/datasource/nuget/__snapshots__/index.spec.ts.snap:1500]: It's the same behavior as in 
`modules/datasource/nuget/index getReleases processes real data (v2) 1`
`modules/datasource/nuget/index getReleases processes real data (v3) feed is not a nuget.org 1`
`modules/datasource/nuget/index getReleases processes real data with no github project url (v2) 1`

**Review by rarkins — APPROVED** [review]

**Comment by NatMarchand** [comment]
Hello!
Thanks for your approval, however there's still a conversation that needs to be resolved if it's ok for you !
Thanks

**Pull request merged**